lib/scrabble_class.rb

```diff
@@ -0,0 +1,65 @@
+module Scrabble
+
+  class Scrabble
+    # SCORE is constant within the class-- tiles always have the same value
+    SCORE = {"a" => 1, "e" => 1, "i" => 1, "o" => 1, "u" => 1,
+             "l" => 1, "n" => 1, "r" => 1, "s" => 1, "t" => 1,
+             "d" => 2, "g" => 2, "b" => 3, "c" => 3, "m" => 3,
+             "p" => 3, "f" => 4, "h" => 4, "v" => 4, "w" => 4,
+             "y" => 4, "k" => 5, "j" => 8, "x" => 8, "q" =>10,
+             "z" => 10}
+
+
+    def self.score(word)
+      total_points = 0
+      # check to see that the input is actually a string (not nil or a number)
+      raise ArgumentError.new "You can only use strings" unless word.class == String
+
+      # check that the string only contains valid English letters, no numbers or symbols, controls for case
+      word_array = word.chars.map { |each| each.downcase}
+      raise ArgumentError.new "You can only use standard English letters" unless word_array.all? { |letter| SCORE[letter] }
+
+      # check that the string is between 1 and 7 letters
+      raise ArgumentError.new "You must play between 1 and 7 letters" unless word_array.length >= 1 && word_array.length <= 7
+
+      # Finally a valid word! Assigning a point value to each letter in the array
+      word_array.each do |letter|
+        points = SCORE[letter]
+        # adds points for each letter to a total_points variable
+        total_points += points
+      end
+      return total_points
+    end
+
+
+    def self.highest_score_from(array_of_words)
+    # transforms array_of_words into words_hash where key == word and value == word score
+    words_hash = Hash[array_of_words.map {|word, total_points| [word, self.score(word)]}]
```

Comment:
OK, first of all: This is good.

A thought experiment: How might you accomplish this without using a ```Hash``, or perhaps using a different ```key``` for the Hash?